### DISCUSSION TITLE
How to use regexManagers:dockerfileVersions?

### DISCUSSION THREAD

**Discussion by dudicoco** [renovatebot/renovate#12716] [Request Help]
Hi,

I'm trying to use https://docs.renovatebot.com/presets-regexManagers/ but i'm not sure how this feature works.

I've added the example from the docs to my `renovate.json` file.

Contents of `Dockerfile`:
```dockerfile
FROM alpine:3.14.3

ENV TERM=xterm-256color

ARG KUBEDOG_VERSION=v0.5.1

RUN wget --no-verbose https://storage.yandexcloud.net/kubedog/targets/releases/${KUBEDOG_VERSION}/kubedog-linux-amd64-${KUBEDOG_VERSION} &&\
  mv kubedog-linux-amd64-${KUBEDOG_VERSION} /usr/bin/kubedog && chmod +x /usr/bin/kubedog

ENTRYPOINT ["/usr/bin/kubedog"]

CMD ["--help"]
```

I thought renovate would bump `KUBEDOG_VERSION` but it doesn't, which I guess makes sense as I assume it would need the entire URL in order to figure out the next version.

Can someone shed some light on how this feature works exactly?

Thanks

**Comment by rarkins** [comment]
It requires an annotation above the ARG line. You can seem some similar examples in https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile

**Reply by dudicoco** [reply]
Thanks @rarkins.

It would be helpful if this was documented, there is also no documentation within https://docs.renovatebot.com/modules/datasource/#github-releases-datasource

I've tried adding the following to my `Dockerfile`:
```Dockerfile
# renovate: datasource=github-releases lookupName=werf/kubedog
ARG KUBEDOG_VERSION=v0.5.1
```

But nothing happened, I also don't see anything in the renovate logs.

Any ideas?

**Reply by viceice** [reply]
You need depName too, lookupName is optional

**Reply by rarkins** [reply]
@viceice do you recall if we have any examples of this type of regex annotation left in public repos?

**Reply by viceice** [reply]
Yes, i think in buildpack are some samples

**Reply by dudicoco** [reply]
Thanks @rarkins and @viceice.

It works now with `depNames`, the example in https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile is incorrect as it uses `lookupName`.
After searching the website i've found some helpful documentation in https://docs.renovatebot.com/modules/manager/regex/#advanced-capture, it would be helpful for other people if there was a link to this page from the other pages mentioned in this discussion.

I'm running into a different problem now - renovate has opened a PR to bump the version, however the build fails now as that github release does not contain any downloads: https://github.com/werf/kubedog/releases/tag/v0.6.2

Is there a way to make renovate check if the release contains downloads?

**Reply by viceice** [reply]
https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile#L1-L2 is the sample you should look at

Those are another custom regex manager for our own buildpack, here the depName is parsed from L61.
https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile#L60-L61
https://github.com/renovatebot/.github/blob/1cfabab3839113ac37eacb27605cb26757db5eeb/default.json#L121

**Reply by dudicoco** [reply]
@viceice can you advise regarding my other question - is there a way to make renovate check if the release contains downloads?

**Reply by rarkins** [reply]
No, there is not
